wa/workloads/googlemaps/uiauto/app/src/main/java/com/arm/wa/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,149 @@
+package com.arm.wa.uiauto.googlemaps;
+
+import android.app.Activity;
+import android.os.Bundle;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import android.support.test.runner.AndroidJUnit4;
+
+import android.util.Log;
+import android.view.KeyEvent;
+
+// Import the uiautomator libraries
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiScrollable;
+import android.support.test.uiautomator.UiSelector;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    protected Bundle parameters;
+    protected String packageID;
+    protected boolean offlineMode;
+
+    public static String TAG = "googlemaps";
+
+    @Before
+    public void initialize() throws Exception {
+        parameters = getParams();
+        packageID = getPackageID(parameters);
+        offlineMode = parameters.getBoolean("offline_mode");
+    }
+
+    @Test
+    public void setup() throws Exception {
+        setScreenOrientation(ScreenOrientation.NATURAL);
+        runApplicationSetup();
+    }
+
+    public void search(String query, String box) throws Exception {
+        UiObject searchBox = mDevice.findObject(new UiSelector().resourceId(packageID + box)
+                                                                .className("android.widget.EditText"));
+        if (!searchBox.waitForExists(uiAutoTimeout)) {
+            throw new UiObjectNotFoundException("Could not find search box.");
+        }
+        searchBox.click();
+
+        UiObject searchText = mDevice.findObject(new UiSelector().resourceId(packageID + "search_omnibox_edit_text")
+                                                                 .className("android.widget.EditText"));
+        searchText.click();
+        searchText.setText(query);
+    }
+
+    public void selectSearchResultContaining(String str) throws Exception {
+        UiObject match = mDevice.findObject(new UiSelector().textContains(str)
+                                                            .className("android.widget.TextView"));
+        if (!match.waitForExists(uiAutoTimeout)) {
+            throw new UiObjectNotFoundException("Could not find search result containing \"" + str + "\".");
+        }
+        match.click();
+    }
+
+    public void getDirectionsFromLocation() throws Exception {
+        UiObject directions = mDevice.findObject(new UiSelector().resourceId(packageID + "placepage_directions_button"));
+        directions.clickAndWaitForNewWindow(uiAutoTimeout);
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
```

Comment:
Please move he high-level methods corresponding to a particular stage above the method implementing specific actions.